### PR TITLE
Fix CoalesceBatches elasped_compute metric

### LINKED ISSUE #3894: CoalesceBatches doesn't provide correct elapsed_compute info in metrics
**Describe the bug**
elapsed_compute for `CoalesceBatches` consists of all the time it took to wait for the batch + process it instead only the computation time
 
**To Reproduce**
Look at metrics, see metric suggests to consume a lot more than other nodes (where it doesn´t).

**Expected behavior**
Should measure the computation time.

**Additional context**

### PR DESCRIPTION
**Pull request by Jefffrey**
# Which issue does this PR close?

Closes #3894 (and addresses part of #3494)

# Rationale for this change

# What changes are included in this PR?

Moving start of timer for elasped_compute to after input_batch has been polled for more accurate time

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/core/src/physical_plan/coalesce_batches.rs [hunk 1]
```diff
@@ -197,14 +197,14 @@ impl CoalesceBatchesStream {
...
         let cloned_time = self.baseline_metrics.elapsed_compute().clone();
-        // records time on drop
-        let _timer = cloned_time.timer();
 
         if self.is_closed {
             return Poll::Ready(None);
         }
         loop {
             let input_batch = self.input.poll_next_unpin(cx);
+            // records time on drop
```
Comment (alamb) [inline datafusion/core/src/physical_plan/coalesce_batches.rs:206]: 👍

**Review by alamb — APPROVED** [review]
Thank you @Jefffrey 

cc @waynexia @Dandandan  @NGA-TRAN

**Review by Dandandan — APPROVED** [review]

**Comment by Dandandan** [comment]
Thanks @Jefffrey !

File: datafusion/core/src/physical_plan/coalesce_batches.rs (hunk 1 above)
Comment (waynexia) [reply datafusion/core/src/physical_plan/coalesce_batches.rs:206]: good job 👍

**Pull request merged**